main.cc

```diff
@@ -1956,6 +1956,12 @@ To start the scylla server proper, simply invoke as: scylla server (or just scyl
                 return std::chrono::duration_cast<steady_clock_type::duration>(std::chrono::milliseconds(cfg->service_levels_interval()));
             }, ss.local(), group0_client);
 
+            // Initialize virtual table in system_distributed keyspace after joining the cluster, so
+            // that the keyspace is ready
+            view_builder.invoke_on_all([] (db::view::view_builder& vb) {
+                vb.init_virtual_table();
+            }).get();
```

Comment:
Can't we do this when starting the view builder itself?

Reply:
I think the problem is that the distributed table is not ready yet when view builder is started, so I added this here after joining the cluster and starting sys_dist_ks

Reply:
I see.